spec/models/league_spec.rb

```diff
@@ -140,6 +141,146 @@
       end
     end
+
+    context 'swiss system' do
+      let(:gamemode) { League.game_modes[:swiss] }
+
+      it 'does not have an up-to-date schedule if the last gameday is in the past' do
+        league.gamedays << FactoryBot.create(:gameday)
+
+        last_gameday = league.gamedays.last
+        last_gameday.starttime = 2.days.ago
+        last_gameday.endtime = 1.day.ago
+
+        expect(league.is_up_to_date).to be false
+      end
+
+      it 'has an up-to-date schedule if the last gameday is not over yet' do
+        league.gamedays << FactoryBot.create(:gameday)
+
+        last_gameday = league.gamedays.last
+        last_gameday.starttime = Date.today.next_day
+        last_gameday.endtime = Date.today.next_day 2
+
+        expect(league.is_up_to_date).to be true
+      end
+
+      it 'allows to check if two teams have already played a match' do
+        teams = FactoryBot.create_list(:team, 2)
+
+        expect(league.have_already_played(teams[0], teams[1])).to be false
+        expect(league.have_already_played(teams[1], teams[0])).to be false
+
+        league.matches << Match.new(team_home: teams[0], team_away: teams[1], gameday_number: 0)
+
+        expect(league.have_already_played(teams[0], teams[1])).to be true
+        expect(league.have_already_played(teams[1], teams[0])).to be true
+      end
+
+      context 'update schedule' do
+        it 'creates a new gameday' do
+          gameday_amount = league.gamedays.length
+          league.update_schedule
+          expect(league.gamedays.length).to be gameday_amount + 1
+        end
+
+
+        it 'creates a correct amount of matches' do
+          matches_amount = league.matches.length
+          league.update_schedule
+          expect(league.matches.length).to be matches_amount + (league.teams.length / 2).floor
+        end
+
+        it 'does not create matches of teams that already played against each other' do
+          league.update_schedule
+          current_pairings = league.matches.map { |match| Set[match.team_home, match.team_away] }
+          expect(current_pairings.uniq.length == current_pairings.length).to be true
+        end
+
+        it 'creates the matches correctly by the ranking' do
+          league = FactoryBot.create(:league)
+          league.game_mode = League.game_modes[:swiss]
+          teams = FactoryBot.create_list(:team, 6)
+          league.teams.append teams
+          league.gamedays << FactoryBot.create(:gameday)
+
+          match1 = FactoryBot.create(:match, team_home: teams[0], team_away: teams[1], gameday_number: 1,
+                                      points_home: 3, points_away: 0)
+          match1.game_results << FactoryBot.build(:game_result,
+                                                  score_home: 20, # team 0
+                                                  score_away: 0   # team 1
+          )
+          league.matches << match1
```

Comment:
If you are using gamedays then there is no need to add the match manually to `league.matches`. Just add it to the matches of the gameday and refer to matches in your code via `league.all_matches`.

Reply:
Where is the difference between calling `league.matches << match1` and `gameday.matches << match1`?

Reply:
ideally at some point we can remove the matches association from events when every feature uses gamedays instead. So I'd prefer it if the tests also do so, because that we can actually clean those relations up.

Reply:
Ah yeah alright.